### ISSUE TITLE
[assimp x64-linux] build failure

### ISSUE THREAD

**Issue by pweissenbacher** [microsoft/vcpkg#8806]
**Host Environment**
- OS: Ubuntu 18.04.3 LTS
- Compiler: G++ 7.4.0, CMake 3.10.2

**To Reproduce**
- Setting up (installing if needed) VCPKG - see SetupVCPKG.txt (Originally is a cmake file included in main CMakeLists.txt) 
- Installing Dependencies with:
```
execute_process(COMMAND ${VCPKG_EXEC} install ${MY_PROJECT_DEPENDENCIES} --triplet ${VCPKG_TARGET_TRIPLET} WORKING_DIRECTORY ${VCPKG_ROOT})
```

**Failure logs** 
```
[191/221] /usr/bin/c++ -DASSIMP_BUILD_DLL_EXPORT -DASSIMP_BUILD_NO_C4D_IMPORTER -DASSIMP_BUILD_NO_OWN_ZLIB -DASSIMP_IMPORTER_GLTF_USE_OPEN3DGC=1 -DMINIZ_USE_UNALIGNED_LOADS_AND_STORES=0 -DOPENDDLPARSER_BUILD -Iinclude -I. -I/home/ubuntu/vcpkg_cpptruths/buildtrees/assimp/src/v5.0.0-c7e52001a1/include -I/home/ubuntu/vcpkg_cpptruths/buildtrees/assimp/src/v5.0.0-c7e52001a1/code -I/home/ubuntu/vcpkg_cpptruths/buildtrees/assimp/src/v5.0.0-c7e52001a1/. -I/home/ubuntu/vcpkg_cpptruths/installed/x64-linux/include -I/home/ubuntu/vcpkg_cpptruths/buildtrees/assimp/src/v5.0.0-c7e52001a1/code/../contrib/rapidjson/include -I/home/ubuntu/vcpkg_cpptruths/buildtrees/assimp/src/v5.0.0-c7e52001a1/code/../contrib -I/home/ubuntu/vcpkg_cpptruths/buildtrees/assimp/src/v5.0.0-c7e52001a1/code/../contrib/unzip -I/home/ubuntu/vcpkg_cpptruths/buildtrees/assimp/src/v5.0.0-c7e52001a1/contrib/irrXML -I/home/ubuntu/vcpkg_cpptruths/buildtrees/assimp/src/v5.0.0-c7e52001a1/code/../contrib/openddlparser/include -I/home/ubuntu/vcpkg_cpptruths/buildtrees/assimp/src/v5.0.0-c7e52001a1/code/../include -Icode/../include -I/home/ubuntu/vcpkg_cpptruths/buildtrees/assimp/src/v5.0.0-c7e52001a1/code/.. -g -fvisibility=hidden -fno-strict-aliasing -Wall -fPIC -std=c++0x -fPIC -D_CRT_SECURE_NO_WARNINGS -O3 -DNDEBUG -MD -MT code/CMakeFiles/assimp.dir/FBX/FBXExporter.cpp.o -MF code/CMakeFiles/assimp.dir/FBX/FBXExporter.cpp.o.d -o code/CMakeFiles/assimp.dir/FBX/FBXExporter.cpp.o -c /home/ubuntu/vcpkg_cpptruths/buildtrees/assimp/src/v5.0.0-c7e52001a1/code/FBX/FBXExporter.cpp
```
[install-x64-linux-rel-out.log](https://github.com/microsoft/vcpkg/files/3787602/install-x64-linux-rel-out.log)

**Additional context**
Build runs successfully on windows (x64-windows). Tried updating vcpkg (./vcpkg update) and reinstalling vcpkg. Clone the recent version of the vcpkg source code from Git.

**Comment by JackBoosY** [comment]
Hi @pweissenbacher, if you use cmake in your custom project, you can copy the dependencies without using `execute_process`. Here is the integration documentation:
https://github.com/microsoft/vcpkg/blob/master/docs/users/integration.md
which you can use in your work environment.

**Comment by pweissenbacher** [comment]
My problem is, that i need to deploy my custom Project which depends on several dependencies (GLFW, Assimp, ...) 
My build runs on a Pipeline (each build runs gets a resetted build machine), where on each run of the build, VCPKG and all dependency packages need to be installed. And the installation of the assimp package gets stuck.

Maybe i got a really big twist in my thinking here, but i don't think that the `vcpkg integrate install` will help here.

**Comment by JackBoosY** [comment]
There is no error in your provisioning log, but I found that the compilation process was interrupted:
```
[191/221] /usr/bin/c++  -DASSIMP_BUILD_DLL_EXPORT -DASSIMP_BUILD_NO_C4D_IMPORTER -DASSIMP_BUILD_NO_OWN_ZLIB -DASSIMP_IMPORTER_GLTF_USE_OPEN3DGC=1 -DMINIZ_USE_UNALIGNED_LOADS_AND_STORES=0 -DOPENDDLPARSER_BUILD -Iinclude -I. -I/home/ubuntu/vcpkg_cpptruths/buildtrees/assimp/src/v5.0.0-c7e52001a1/include -I/home/ubuntu/vcpkg_cpptruths/buildtrees/assimp/src/v5.0.0-c7e52001a1/code -I/home/ubuntu/vcpkg_cpptruths/buildtrees/assimp/src/v5.0.0-c7e52001a1/. -I/home/ubuntu/vcpkg_cpptruths/installed/x64-linux/include -I/home/ubuntu/vcpkg_cpptruths/buildtrees/assimp/src/v5.0.0-c7e52001a1/code/../contrib/rapidjson/include -I/home/ubuntu/vcpkg_cpptruths/buildtrees/assimp/src/v5.0.0-c7e52001a1/code/../contrib -I/home/ubuntu/vcpkg_cpptruths/buildtrees/assimp/src/v5.0.0-c7e52001a1/code/../contrib/unzip -I/home/ubuntu/vcpkg_cpptruths/buildtrees/assimp/src/v5.0.0-c7e52001a1/contrib/irrXML -I/home/ubuntu/vcpkg_cpptruths/buildtrees/assimp/src/v5.0.0-c7e52001a1/code/../contrib/openddlparser/include -I/home/ubuntu/vcpkg_cpptruths/buildtrees/assimp/src/v5.0.0-c7e52001a1/code/../include -Icode/../include -I/home/ubuntu/vcpkg_cpptruths/buildtrees/assimp/src/v5.0.0-c7e52001a1/code/.. -g -fvisibility=hidden -fno-strict-aliasing -Wall -fPIC -std=c++0x -fPIC  -D_CRT_SECURE_NO_WARNINGS -O3 -DNDEBUG -MD -MT code/CMakeFiles/assimp.dir/FBX/FBXExporter.cpp.o -MF code/CMakeFiles/assimp.dir/FBX/FBXExporter.cpp.o.d -o code/CMakeFiles/assimp.dir/FBX/FBXExporter.cpp.o -c /home/ubuntu/vcpkg_cpptruths/buildtrees/assimp/src/v5.0.0-c7e52001a1/code/FBX/FBXExporter.cpp
```
Maybe someone interrupted the build process?

And I don't know about your environment. Is the path where vcpkg is located on your project machine? I think there are several solutions here:
1. Install vcpkg on the project computer. Build dependencies once and run `integrate`.
2. Build dependencies using vcpkg, export the dependencies using the command `export`, and copy them into your project to use them.
3. Just like your solution. First use vcpkg to install dependencies and then build your own project.

But I can successfully build assimp, can you try to install assimp manually and provide the results?

Thanks.

**Comment by pweissenbacher** [comment]
@JackBoosY you are indeed right, I myself interrupted the build process, but mainly because the machine were I built, got stuck at package `[191/221]`. I tried to build it manually with the `vcpkg install assimp` command and there I got an error message why the built was not successfull.

The error message was: `virtual memory exhausted: Cannot allocate memory`.
Apparently, 4GB of memory is not enough to uild assimp...
And somehow, the automatic install during the CMake build did not provide the error message.

But thanks for your investigations, I will try to build assimp with a more powerful machine (more memory).

Best Regards!
